pkg/background/mutate/mutate.go

```diff
@@ -84,11 +84,27 @@ func (c *MutateExistingController) ProcessUR(ur *kyvernov1beta1.UpdateRequest) e
 			continue
 		}
 
-		trigger, err := common.GetResource(c.client, ur.Spec, c.log)
-		if err != nil || trigger == nil {
-			logger.WithName(rule.Name).Error(err, "failed to get trigger resource")
-			errs = append(errs, err)
-			continue
+		var trigger *unstructured.Unstructured
+		admissionRequest := ur.Spec.Context.AdmissionRequestInfo.AdmissionRequest
+		if admissionRequest != nil && admissionRequest.SubResource != "" {
+			trigger, err = common.GetResource(c.client, ur.Spec, c.log)
```

Comment:
Can we use the admission request object directly as the trigger? Is it needed to fetch from the cluster?

Reply:
+1: Update requests are put in a queue before execution. In the meantime, if the particular resource changes, then we should fetch the latest version of that resource.

-1: There was some specific condition in that version of the resource which isn't in the new version that we fetched, this is potentially buggy.

From the slack discussion:
>There’s a use case for generate, say create a confimap on the namespace’s creation. For such a scenario, the namespace has to be exist before creating the configmap into that ns.

This applies to `mutate-existing` too, for eg: You mutate a config map on the creation of some resource. You modify the `config-map` to add the `uid` of the resource created. For this the resource needs to be created first, otherwise `uid` will be null.

Reply:
Let's only fetch any type of resource if the request incoming request is of type `CREATE`. For subresources `CREATE`ion, if that particular resource is gettable, we will get it, otherwise, `object`. 

For other cases, we will resort to `object`, if that is nil then `oldObject`.

Reply:
>Let's only fetch any type of resource if the request incoming request is of type CREATE. For subresources CREATEion, if that particular resource is gettable, we will get it, otherwise, object.

>For other cases, we will resort to object, if that is nil then oldObject.

Sounds good.